drivers/video/gc2145.c

```diff
@@ -0,0 +1,1164 @@
+	{0x6d, 0x8b},
+	{0x6e, 0x50},
+	{0x6f, 0x18},
+	{0x73, 0xf0},
+	{0x70, 0x0d},
+	{0x71, 0x60},
+	{0x72, 0x80},
+	{0x74, 0x01},
+	{0x75, 0x01},
+	{0x7f, 0x0c},
+	{0x76, 0x70},
+	{0x77, 0x58},
+	{0x78, 0xa0},
+	{0x79, 0x5e},
+	{0x7a, 0x54},
+	{0x7b, 0x58},
+	{0xfe, 0x00},
+	{0xfe, 0x02},
+	{0xc0, 0x01},
+	{0xc1, 0x44},
+	{0xc2, 0xfd},
+	{0xc3, 0x04},
+	{0xc4, 0xF0},
+	{0xc5, 0x48},
+	{0xc6, 0xfd},
+	{0xc7, 0x46},
+	{0xc8, 0xfd},
+	{0xc9, 0x02},
+	{0xca, 0xe0},
+	{0xcb, 0x45},
+	{0xcc, 0xec},
+	{0xcd, 0x48},
+	{0xce, 0xf0},
+	{0xcf, 0xf0},
+	{0xe3, 0x0c},
+	{0xe4, 0x4b},
+	{0xe5, 0xe0},
+	{0xfe, 0x01},
+	{0x9f, 0x40},
+	{0xfe, 0x00},
+
+	/*	OUTPUT Control  */
+	{0xfe, 0x00},
+	{0xf2, 0x0f},
+	{0xfe, 0x02},
+	{0x40, 0xbf},
+	{0x46, 0xcf},
+	{0xfe, 0x00},
+
+	{0xfe, 0x00},
+	{0x05, 0x01},
+	{0x06, 0x1C},
+	{0x07, 0x00},
+	{0x08, 0x32},
+	{0x11, 0x00},
+	{0x12, 0x1D},
+	{0x13, 0x00},
+	{0x14, 0x00},
+
+	{0xfe, 0x01},
+	{0x3c, 0x00},
+	{0x3d, 0x04},
+	{0xfe, 0x00},
+	{0x00, 0x00},
+};
+
+struct gc2145_config {
+	struct i2c_dt_spec i2c;
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
+};
+
+struct gc2145_data {
+	struct video_format fmt;
+};
+
+#define GC2145_VIDEO_FORMAT_CAP(width, height, format) \
+	{ \
+		.pixelformat = format, \
+		.width_min = width, \
+		.width_max = width, \
+		.height_min = height, \
+		.height_max = height, \
+		.width_step = 0, \
+		.height_step = 0 \
+	}
+
+static const struct video_format_cap fmts[] = {
+	GC2145_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_RGB565), /* UXGA  */
+	{ 0 }
+};
+
+static int gc2145_write_reg(const struct i2c_dt_spec *spec, uint8_t reg_addr, uint8_t value)
+{
+	int ret = 0;
+	uint8_t tries = 3;
+
+	/**
+	 * It rarely happens that the camera does not respond with ACK signal.
+	 * In that case it usually responds on 2nd try but there is a 3rd one
+	 * just to be sure that the connection error is not caused by driver
+	 * itself.
+	 */
+	do {
+		ret = i2c_reg_write_byte_dt(spec, reg_addr, value);
+		if (ret < 0) {
+			LOG_ERR("failed to write 0x%x to 0x%x", value, reg_addr);
+			/* If writing failed wait 5ms before next attempt */
+			k_msleep(5);
+		}
+
+	} while ((tries--) || (ret != 0));
+
+	return ret;
+}
+
+static int gc2145_read_reg(const struct i2c_dt_spec *spec, uint8_t reg_addr, uint8_t *value)
+{
+	int ret = 0;
+	uint8_t tries = 3;
+
+	/**
+	 * It rarely happens that the camera does not respond with ACK signal.
+	 * In that case it usually responds on 2nd try but there is a 3rd one
+	 * just to be sure that the connection error is not caused by driver
+	 * itself.
+	 */
+	do {
+		ret = i2c_reg_read_byte_dt(spec, reg_addr, value);
+		if (ret < 0) {
+			LOG_ERR("failed to read 0x%x register", reg_addr);
+			/* If writing failed wait 5ms before next attempt */
+			k_msleep(5);
+		}
+
+	} while ((tries--) || (ret != 0));
+
+	return ret;
+}
+
+static int gc2145_write_all(const struct device *dev, const struct gc2145_reg *regs,
+			uint16_t reg_num)
```

Comment:
Code must be indented inside the last open parenthesis, like this: 
```suggestion
static int gc2145_write_all(const struct device *dev, const struct gc2145_reg *regs,
			    uint16_t reg_num)
```
Make sure you are using tabs and set the tab width to 8; use tabs then add the minimal amount of spaces required to properly align.